File: storeadmin/api/mutations/listing/updateListing.ts
import { listingFragment } from '@/api/fragments'
import gql from 'graphql-tag'

export default gql`
  mutation updateListing($uuid: ID!, $attributes: ListingAttributes!) {
    updateListing(input: { uuid: $uuid, attributes: $attributes }) {
      listing {
        ...listingFragment
      }
      error
    }
  }
  ${listingFragment}
`

File: server/app/graphql/mutations/listing/update_listing.rb
class Mutations::Listing::UpdateListing < Mutations::BaseMutation
  null false

  argument :uuid, ID, required: true
  argument :attributes, InputTypes::Listing, required: true

  field :listing, ObjectTypes::Listing, null: true
  field :error, String, null: true

  def resolve(uuid:, attributes:)
    listing = ::Listing.find_by!(uuid: uuid)
    listing.update!(attributes.to_h)

    { listing: listing.reload }
  rescue => e
    Rails.logger.error(e)
    return GraphQL::ExecutionError.new(e.message)
  end
end
